payloads/v9/channel.ts

```diff
@@ -0,0 +1,1047 @@
+	/**
+	 * This message came from the urgent message system
+	 */
+	Urgent = 1 << 4,
+	/**
+	 * This message has an associated thread, which shares its id
+	 */
+	HasThread = 1 << 5,
+	/**
+	 * This message is only visible to the user who invoked the Interaction
+	 */
+	Ephemeral = 1 << 6,
+	/**
+	 * This message is an Interaction Response and the bot is "thinking"
+	 */
+	Loading = 1 << 7,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#message-object-message-sticker-structure
+ */
+export interface APISticker {
+	/**
+	 * ID of the sticker
+	 */
+	id: Snowflake;
+	/**
+	 * ID of the pack the sticker is from
+	 */
+	pack_id: Snowflake;
+	/**
+	 * Name of the sticker
+	 */
+	name: string;
+	/**
+	 * Description of the sticker
+	 */
+	description: string;
+	/**
+	 * A comma-separated list of tags for the sticker
+	 */
+	tags?: string;
+	/**
+	 * Sticker asset hash
+	 */
+	asset: string;
+	/**
+	 * Type of sticker format
+	 *
+	 * See https://discord.com/developers/docs/resources/channel#message-object-message-sticker-format-types
+	 */
+	format_type: StickerFormatType;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#message-object-message-sticker-format-types
+ */
+export const enum StickerFormatType {
+	PNG = 1,
+	APNG,
+	LOTTIE,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#followed-channel-object
+ */
+export interface APIFollowedChannel {
+	/**
+	 * Source channel id
+	 */
+	channel_id: Snowflake;
+	/**
+	 * Created target webhook id
+	 */
+	webhook_id: Snowflake;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#reaction-object-reaction-structure
+ */
+export interface APIReaction {
+	/**
+	 * Times this emoji has been used to react
+	 */
+	count: number;
+	/**
+	 * Whether the current user reacted using this emoji
+	 */
+	me: boolean;
+	/**
+	 * Emoji information
+	 *
+	 * See https://discord.com/developers/docs/resources/emoji#emoji-object
+	 */
+	emoji: APIPartialEmoji;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#overwrite-object-overwrite-structure
+ */
+export interface APIOverwrite {
+	/**
+	 * Role or user id
+	 */
+	id: Snowflake;
+	/**
+	 * Either 0 (role) or 1 (member)
+	 *
+	 * {@link OverwriteType}
+	 */
+	type: OverwriteType;
+	/**
+	 * Permission bit set
+	 *
+	 * See https://discord.com/developers/docs/topics/permissions#permissions-bitwise-permission-flags
+	 *
+	 * See https://en.wikipedia.org/wiki/Bit_field
+	 */
+	allow: Permissions;
+	/**
+	 * Permission bit set
+	 *
+	 * See https://discord.com/developers/docs/topics/permissions#permissions-bitwise-permission-flags
+	 *
+	 * See https://en.wikipedia.org/wiki/Bit_field
+	 */
+	deny: Permissions;
+}
+
+export const enum OverwriteType {
+	Role,
+	Member,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#thread-metadata-object-thread-metadata-structure
+ */
+export interface APIThreadMetadata {
+	/**
+	 * The archived state of the thread
```

Comment:
These should use the same strings that the docs use

See https://discord.com/developers/docs/resources/channel#thread-metadata-object